### PR TITLE
Fix #2332

### LINKED ISSUE #2332: JSONDecodeError with PyPy3.8 and isolated_build=True
[Here is the project.](https://github.com/zariiii9003/python-can/tree/b5669ec67a54f87836edae9b571fc648bd304570)

The error occurs with Ubuntu, macOS and Windows, but only with PyPy3.8. CPython and PyPy3.7 works fine. But PyPy3.8 works fine without `isolated_build=True`.

<details open>
<summary>Traceback</summary>

```
  tox -e gh
  shell: /usr/bin/bash -e {0}
  env:
    PY_COLORS: 1
    pythonLocation: /opt/hostedtoolcache/PyPy/3.8.12/x64/bin
.package create: /home/runner/work/python-can/python-can/.tox/.package
.package installdeps: setuptools >= 40.8, wheel
Traceback (most recent call last):
  File "/opt/hostedtoolcache/PyPy/3.8.12/x64/bin/tox", line 8, in <module>
    sys.exit(cmdline())
  File "/opt/hostedtoolcache/PyPy/3.8.12/x64/lib/pypy3.8/site-packages/tox/session/__init__.py", line 44, in cmdline
    main(args)
  File "/opt/hostedtoolcache/PyPy/3.8.12/x64/lib/pypy3.8/site-packages/tox/session/__init__.py", line 69, in main
    exit_code = session.runcommand()
  File "/opt/hostedtoolcache/PyPy/3.8.12/x64/lib/pypy3.8/site-packages/tox/session/__init__.py", line 197, in runcommand
    return self.subcommand_test()
  File "/opt/hostedtoolcache/PyPy/3.8.12/x64/lib/pypy3.8/site-packages/tox/session/__init__.py", line 213, in subcommand_test
    venv.package = self.hook.tox_package(session=self, venv=venv)
  File "/opt/hostedtoolcache/PyPy/3.8.12/x64/lib/pypy3.8/site-packages/pluggy/_hooks.py", line 265, in __call__
    return self._hookexec(self.name, self.get_hookimpls(), kwargs, firstresult)
  File "/opt/hostedtoolcache/PyPy/3.8.12/x64/lib/pypy3.8/site-packages/pluggy/_manager.py", line 80, in _hookexec
    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
  File "/opt/hostedtoolcache/PyPy/3.8.12/x64/lib/pypy3.8/site-packages/pluggy/_callers.py", line 60, in _multicall
    return outcome.get_result()
  File "/opt/hostedtoolcache/PyPy/3.8.12/x64/lib/pypy3.8/site-packages/pluggy/_result.py", line 60, in get_result
    raise ex[1].with_traceback(ex[2])
  File "/opt/hostedtoolcache/PyPy/3.8.12/x64/lib/pypy3.8/site-packages/pluggy/_callers.py", line 39, in _multicall
    res = hook_impl.function(*args)

  File "/opt/hostedtoolcache/PyPy/3.8.12/x64/lib/pypy3.8/site-packages/tox/package/__init__.py", line 16, in tox_package
    session.package, session.dist = get_package(session)
  File "/opt/hostedtoolcache/PyPy/3.8.12/x64/lib/pypy3.8/site-packages/tox/package/__init__.py", line 29, in get_package
    package = acquire_package(config, session)
  File "/opt/hostedtoolcache/PyPy/3.8.12/x64/lib/pypy3.8/site-packages/tox/package/__init__.py", line 40, in acquire_package
    path = build_package(config, session)
  File "/opt/hostedtoolcache/PyPy/3.8.12/x64/lib/pypy3.8/site-packages/tox/package/builder/__init__.py", line 9, in build_package
    return build(config, session)
  File "/opt/hostedtoolcache/PyPy/3.8.12/x64/lib/pypy3.8/site-packages/tox/package/builder/isolated.py", line 37, in build
    build_requires = get_build_requires(build_info, package_venv, config.setupdir)
  File "/opt/hostedtoolcache/PyPy/3.8.12/x64/lib/pypy3.8/site-packages/tox/package/builder/isolated.py", line 149, in get_build_requires
    return json.loads(result.split("\n")[-2])
  File "/opt/hostedtoolcache/PyPy/3.8.12/x64/lib/pypy3.8/json/__init__.py", line 364, in loads
    if _pypyjson else _default_decoder.decode(s))
json.decoder.JSONDecodeError: Unexpected 'w': line 1 column 3 (char 2)
```

</details>

```
distlib-0.3.4 
filelock-3.4.2 
packaging-21.3 
platformdirs-2.4.1 
pluggy-1.0.0 
py-1.11.0 
pyparsing-3.0.7 s
ix-1.16.0 
toml-0.10.2 
tox-3.24.5 
virtualenv-20.13.0
```

### PR DESCRIPTION
**Pull request by zariiii9003**
Fixes #2332

build_requires.py caught a setuptools warning so `json.loads()` failed in isolated.py.
I disabled the capturing of stderr.

## Contribution checklist:

(also see [CONTRIBUTING.rst](../tree/master/CONTRIBUTING.rst) for details)

- [x] wrote descriptive pull request text
- [ ] added/updated test(s)
- [ ] updated/extended the documentation
- [x] added relevant [issue keyword](https://help.github.com/articles/closing-issues-using-keywords/)
      in message body
- [x] added news fragment in [changelog folder](../tree/master/docs/changelog)
  * fragment name: `<issue number>.<type>.rst` for example (588.bugfix.rst)
  * `<type>` is must be one of `bugfix`, `feature`, `deprecation`, `breaking`, `doc`, `misc`
  * if PR has no issue: consider creating one first or change it to the PR number after creating the PR
  * "sign" fragment with ```-- by :user:`<your username>`.```
  * please, use full sentences with correct case and punctuation, for example:
    ```rst
    Fixed an issue with non-ascii contents in doctest text files -- by :user:`superuser`.
    ```
  * also see [examples](../tree/master/docs/changelog)
- [ ] added yourself to `CONTRIBUTORS` (preserving alphabetical order)

### CODE REVIEW

File: src/tox/package/builder/isolated.py [hunk 1]
```diff
@@ -145,5 +145,6 @@ def get_build_requires(build_info, package_venv, setup_dir):
             returnout=True,
             action=action,
             cwd=setup_dir,
+            capture_err=False,
         )
-        return json.loads(result.split("\n")[-2])
+        return json.loads(result.strip().split("\n")[-1])
```
Comment (gaborbernat) [inline src/tox/package/builder/isolated.py:150]: Why do we need strip here?

**Review by gaborbernat — CHANGES_REQUESTED** [review]
Let's add a test here. And a changelog should go into a news fragment file, not directly to the changelog one 👍

File: src/tox/package/builder/isolated.py (hunk 1 above)
Comment (zariiii9003) [reply src/tox/package/builder/isolated.py]: I removed it. Let's keep changes minimal.

**Review by gaborbernat — COMMENTED** [review]
Can we test this change somehow in a unit test?

**Comment by zariiii9003** [comment]
> Can we test this change somehow in a unit test?

Tbh i have zero knowledge about the inner workings of this library and i don't know how to test this.
I don't even understand why the results are read from the stdout/stderr instead of a simple function return value.

**Pull request merged**